### DISCUSSION TITLE
.bb file highlighted when viewed directly, but not in PR diff

### DISCUSSION THREAD

**Discussion by ghost** [github-linguist/linguist#6064] [General]
I’m not sure what category this belongs in.

I have a Pull Request in a private repo that makes some changes to a file with a `.bb` extension, for [Babashka](https://github.com/babashka/babashka), a Clojure interpreter.

The puzzling behavior I’m seeing is: when I navigate directly to the file (via a `blob` URL) the syntax is highlighted correctly. But in the PR diff, the syntax is not highlighted at all — it’s all one color, seems like black.

I’m puzzled because I found [the entry for Clojure](https://github.com/github/linguist/blob/bf853f1c663903e3ee35935189760191f1c45e1c/lib/linguist/languages.yml#L1070-L1092) in `languages.yml` and it does include the `.bb` extension.

Is this expected behavior?

If it matters, the file starts with:

```text
#!/usr/bin/env bb
#_{:clj-kondo/ignore [:namespace-name-mismatch]}
(ns my-namespace
  (:require [etc etc]))
```

Maybe there are some heuristics at play that look at the first line of the file to help inform what kind of `.bb` file it is? But if so, why would those heuristics work in one context but not the other?

**Comment by lildude** [comment]
AFAIK this should work. I've done a quick test and syntax highlighting works in the PR diff.

Are you able to reproduce this in a public repo so I can see it in action?

**Reply by ghost** [reply]
Thanks for the rapid assistance!

I’ve just opened [a PR in a public repo](https://github.com/latacora/gh-actions/pull/2/) that I thought would reproduce the issue, but it hasn’t. I even tried adding the directive `#_{:clj-kondo/ignore [:namespace-name-mismatch]}` above the `ns` form but that didn’t trigger the problem either.

So — I dunno. I did double-check the original PR that exhibited the problem, and it is still exhibiting the problem. I believe it’s happening for me, using Safari, and for my co-worker, using Chrome (both on Mac).

🤷

**Reply by lildude** [reply]
Check your browser devtools console... the syntax highlighting is added asynchronously so maybe something's blocking it or it's hitting problems which might be reported to the console.

Unfortunately, this side of things is completely independent from Linguist so if you're still seeing the issue and there's nothing obvious in the web console, I recommend contacting GitHub support using the "Contact GitHub" link at the bottom of the affected page and they can look into it.

**Reply by ghost** [reply]
Makes sense — thank you!
